server/tournaments/index.ts

```diff
@@ -1831,15 +2126,66 @@ const commands: ChatCommands = {
 			`- announce/announcements &lt;on|off>: Enables/disables tournament announcements for the current room.<br />` +
 			`- banuser/unbanuser &lt;user>: Bans/unbans a user from joining tournaments in this room. Lasts 2 weeks.<br />` +
 			`- sub/replace &lt;olduser>, &lt;newuser>: Substitutes a new user for an old one<br />` +
+			`- settings: Do <code>/help tour settings</code> for more information<br />` +
 			`More detailed help can be found <a href="https://www.smogon.com/forums/threads/3570628/#post-6777489">here</a>`
 		);
 	},
 };
 
+const roomSettings: SettingsHandler[] = [
+	room => ({
+		label: "Tournament Forced Public Battles",
+		permission: "editroom",
+		options: [
+			['on', room.settings.tournaments?.forcePublic || 'tour settings forcepublic on'],
+			['off', !room.settings.tournaments?.forcePublic || 'tour settings forcepublic off'],
+		],
+	}),
+	room => ({
+		label: "Tournament Forced Timer",
+		permission: "editroom",
+		options: [
+			['on', room.settings.tournaments?.forceTimer || 'tour settings forcetimer on'],
+			['off', !room.settings.tournaments?.forceTimer || 'tour settings forcetimer off'],
+		],
+	}),
+	room => ({
+		label: "Tournament Modjoin",
+		permission: "editroom",
+		options: [
+			['allow', room.settings.tournaments?.allowModjoin || 'tour settings modjoin allow'],
+			['disallow', !room.settings.tournaments?.allowModjoin || 'tour settings modjoin disallow'],
+		],
+	}),
+	room => ({
+		label: "Tournament Scouting",
+		permission: "editroom",
+		options: [
+			['allow', room.settings.tournaments?.allowScouting || 'tour settings scouting allow'],
+			['disallow', !room.settings.tournaments?.allowScouting || 'tour settings scouting disallow'],
+		],
+	}),
+];
+
 export const Tournaments = {
 	TournamentGenerators,
 	TournamentPlayer,
 	Tournament,
 	createTournament,
 	commands,
+	roomSettings,
 };
+
+for (const room of Rooms.rooms.values()) {
+	// @ts-ignore
+	const announcements = room.settings.tourAnnouncements;
+	// @ts-ignore
+	delete room.settings.tourAnnouncements;
```

Comment:
```suggestion
	const announcements = (room.settings as any).tourAnnouncements;
	delete (room.settings as any).tourAnnouncements;
```

I'd prefer `any`-casting to `ts-ignore`, going forward.

(The reason is that `any`-casting only suppresses some errors, while `ts-ignore` suppresses every error, maybe including some actual bugs.)